usvm-ts-dataflow/src/main/kotlin/org/usvm/dataflow/ts/infer/TypeInferenceManager.kt

```diff
@@ -380,6 +375,7 @@ class TypeInferenceManager(
                 .toList()
 
             val typeFactsOnThisCtor = forwardSummariesByClass[cls.signature].orEmpty()
+                .asSequence()
```

Comment:
Consider improving the performance even more by removing `.distinct().toList()` below. We already do this when combining `typeFactsOnThis`.
Do not forget to mark both sequences `.constrainOnce()`.